### PR TITLE
Wrong skip criterion in tests/output-complete-obj/test.ml

### PR DESCRIPTION
**Pull request by dra27**
I didn't check the history, but I think this test was using `libunix` to disable this test on Windows - we have a test for that called `not-windows`! Changing to `not-windows` means that the test gets correctly run on the `MIN_BUILD=1` Travis as well, since the test doesn't depend on Unix at all.

All that said, this test should be updated to run correctly on Windows, which is why it's draft for now.

### CODE REVIEW

**Comment by shindere** [comment]
David Allsopp (2020/07/28 04:39 -0700):
> I didn&#39;t check the history, but I think this test was using `libunix` to disable this test on Windows - we have a test for that called `not-windows`! Changing to `not-windows` means that the test gets correctly run on the `MIN_BUILD=1` Travis as well, since the test doesn&#39;t depend on Unix at all.
>
> All that said, this test should be updated to run correctly on
> Windows, which is why it&#39;s draft for now.

So you don't want this PR to be approved right now, do you?
>
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/ocaml/ocaml/pull/9809
>
> -- Commit Summary --
>
>   * Incorrect ocamltest used
>
> -- File Changes --
>
>     M testsuite/tests/output-complete-obj/test.ml (2)
>
> -- Patch Links --
>
> https://github.com/ocaml/ocaml/pull/9809.patch
> https://github.com/ocaml/ocaml/pull/9809.diff
>

**Comment by dra27** [comment]
Not for now, no - but I intend to return to it in the near future, as I think it would be better to use `not-windows` for "really doesn't run on Windows" rather than "I couldn't figure out how to get this to work on Windows, and it was reasonable for it not to block the PR". Part of the purpose in putting the draft up was to be contradicted in my diagnosis of why it had `libunix` (if necessary!)

**Comment by dra27** [comment]
This test was failing in #10135 since it had never had to support bootstrapped flexlink before. That's fixed in #10135, but while I was there I checked what was going on with Windows: this test will succeed when:

- [x] #10135 is merged, which allows `${mkexe}` to be overridden in the same way as `${mkdll}` already is from #1827
- [x] #10376 which fixes `-output-complete-obj` on msvc (this test precisely fails on msvc64 because `-output-complete-obj` wasn't quite doing the complete part!)
- [x] #10377 allowing ocamltest to call `${mkexe}` correctly on mingw to pass `-municode` to the linker properly.

The only change is to remove the "trick" `libunix` test and add `wmain` to the C stub.

**Review by shindere — APPROVED** [review]

**Comment by shindere** [comment]
Nice, thank you.

Let's merge this once #10135 is in.

**Pull request merged**
